sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/interfaces.scala

```diff
@@ -621,6 +621,17 @@ abstract class TypedImperativeAggregate[T] extends ImperativeAggregate {
     buffer(mutableAggBufferOffset) = serialize(getBufferObject(buffer))
   }
 
+  /**
+   * In-place replaces SparkSQL internally supported underlying storage format (BinaryType),
+   * with the aggregation buffer object stored at buffer's index `mutableAggBufferOffset`.
```

Comment:
`stored at buffer's index mutableAggBufferOffset` describes the binary, should be put before `with the aggregation buffer object ...`